pdc_client/plugins/product_version.py

```diff
@@ -0,0 +1,119 @@
+
+        # CRUD: CREATE
+        create_parser = self.add_action('create', help='create new product_version')
+        self.add_product_version_arguments(create_parser, required=True)
+        create_parser.set_defaults(func=self.product_version_create)
+
+        # CRUD: READ one (info)
+        info_parser = self.add_action('info', help='display details of a product_version')
+        info_parser.add_argument('product_version_id', metavar="PRODUCT_VERSION_ID")
+        info_parser.set_defaults(func=self.product_version_info)
+
+        # CRUD: READ many (list)
+        list_parser = self.add_action('list', help='list all product_versions')
+        list_parser.add_argument('--inactive', action='store_true',
+                                 help='show only inactive product_versions')
+        list_parser.add_argument('--all', action='store_true',
+                                 help='show both active and inactive product_versions')
+        list_parser.set_defaults(func=self.product_version_list)
+
+        # CRUD: UPDATE
+        update_parser = self.add_action('update', help='update an existing product_version')
+        update_parser.add_argument('product_version_id')
+        self.add_product_version_arguments(update_parser)
+        update_parser.set_defaults(func=self.product_version_update)
+
+        # CRUD: DELETE
+        # not implemented
+
+    def add_product_version_arguments(self, parser, required=False):
+        required_args = {
+            'short': {},
+            'name': {},
+            'version': {},
+            'product': {},
+        }
+        optional_args = {
+        }
+        add_create_update_args(parser, required_args, optional_args, required)
+
+    def prep_for_print(self, record):
+        result = OrderedDict()
+        result["Product Version ID"] = record["product_version_id"]
+        result["Short Name"] = record["short"]
+        result["Name"] = record["name"]
+        result["Product"] = record["product"]
+        result["Active"] = record["active"] and "yes" or "no"
+        return result
+
+    def product_version_create(self, args):
+        data = self.get_product_version_data(args)
+        self.logger.debug('Creating product_version with data {0}'.format(data))
+        response = self.client["product-versions"]._(data)
+        self.product_version_info(args, response['product_version_id'])
+
+    def product_version_info(self, args, product_version_id=None):
+        product_version_id = product_version_id or args.product_version_id
+        product_version = self.client["product-versions"][product_version_id]._()
+
+        fmt = '{0:20} {1}'
+        for key, value in self.prep_for_print(product_version).items():
+            print(fmt.format(key, value))
+
+    def product_version_list(self, args):
+        filters = {}
+        if args.inactive:
+            filters['active'] = False
+        elif not args.all:
+            filters['active'] = True
```

Comment:
I think both the two ways are okay, so no need change the original codes, what's your thinking?